dotCMS/src/main/java/com/dotmarketing/business/VersionableAPIImpl.java

```diff
@@ -543,22 +541,23 @@ public void setWorking(final Versionable versionable) throws DotDataException, D
         if(identifier.getAssetType().equals("contentlet")) {
 
             final Contentlet contentlet = (Contentlet)versionable;
-            ContentletVersionInfo info = versionableFactory
+            Optional<ContentletVersionInfo> info = versionableFactory
                     .getContentletVersionInfo(contentlet.getIdentifier(), contentlet.getLanguageId());
 
-            if(info ==null || !UtilMethods.isSet(info.getIdentifier())) {
+            if(!info.isPresent()) {
```

Comment:
Shouldn't we check also for `info.getIdentifier()`is set?

Reply:
Good review here